### ISSUE TITLE
Invalid parsing of macro with pattern syntax

### ISSUE THREAD

**Issue by jan-auer** [rust-analyzer/rust-analyzer#6972]
IIRC with 0.2.408 (2020-12-07), one of our macros started emitting "unexpected token in input". The most minimal reproduction I could come up with is:

```rust
macro_rules! mv {
    (|$($p:pat),+| $body:expr) => {{
        move |$($p),+| $body
    }};
}

#[test]
fn test_clone() {
    let print = mv!(|suffix| {
        println!("{}", suffix);
    });

    print("world");
}
```

When hovering the closure it prints the following type signature:
```rust
|| -> || -> ()
```

The culprit _seems_ to be the `pat` argument. If we replace it with `p:ident`, Rust Analyzer parses just fine.

**Comment by edwin0cheng** [comment]
It could be fixed in #6959 and it should be in next nightly.

**Comment by jan-auer** [comment]
Thanks, this is indeed a duplicate of #6882
